### ISSUE TITLE
qt5-qmake: broken because LDFLAGS leaked into mkspecs

### ISSUE THREAD

**Issue by pullmoll** [voidlinux/void-packages#3112]
The build environment's `LDFLAGS` leaked into the `mkspecs/qmodule.pri` and `mkspecs/common/gcc-base.conf` files.

Specificially the `-specs=/void-packages/common/environment/configure/gccspecs/hardened-ld` should not be there.

This is due to `srcpkgs/qt5/template` patching these files for the build process and not cleaning them before installing them in the -qmake subpackage.

I'm working on a solution which does not patch the mkspec files in the first place.

**Comment by pullmoll** [comment]
The same issue applies for `qt-4.8.7`, which is where I stumbled on it.
It will take some more time to test the fix for this as well.

**Comment by equalsraf** [comment]
I'm seeing something similar to this issue with Qt 5.6 in /usr/lib/qt5/mkspecs/qmodule.pri which is used when building qt modules, in my case when building qtwebengine

```
QMAKE_CFLAGS = -specs=/void-packages/common/environment/configure/gccspecs/hardened-cc1  -fstack-protector-strong -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe  -DOPENSSL_NO_SSL3_METHOD -g
QMAKE_CXXFLAGS = -specs=/void-packages/common/environment/configure/gccspecs/hardened-cc1 -fstack-protector-strong -D_FORTIFY_SOURCE=2 -mtune=generic -O2 -pipe  -DOPENSSL_NO_SSL3_METHOD -g
QMAKE_LFLAGS = -specs=/void-packages/common/environment/configure/gccspecs/hardened-ld -Wl,-z,relro -Wl,-z,now -Wl,--as-needed -Wl,-z,relro 
```

**Comment by pullmoll** [comment]
The problem stems from xbps-src modifying the LDFLAGS rather than using e.g. a compiler wrapper or the `GCC_SPECS` environment setting, which is available in `gcc-4.9.3_4`.

I don't know yet, if I should try to fix this on the qt5 side, or if we should change the way in which the specs are passed to gcc.

**Comment by pullmoll** [comment]
Issue was solved long ago.
